### PR TITLE
feat(ec2): Implement UserData methods in MultipartUserData

### PR DESCRIPTION
**Pull request by ddneilson**
Implements: https://github.com/aws/aws-cdk/issues/14346

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-ec2/lib/user-data.ts [hunk 1]
```diff
@@ -484,6 +486,16 @@ export class MultipartUserData extends UserData {
     this.addPart(MultipartBody.fromUserData(userData, contentType));
   }
 
+  /**
+   * Adds a multipart part based on a UserData object. This UserData is always added
+   * as a shell script. The UserData added by this method will also be the target
+   * of calls to the `add*Command` methods on this MultipartUserData object.
+   */
+  public addUserDataPartForCommands(userData: UserData) {
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-ec2/lib/user-data.ts:494]: I think I would prefer the first invocation of `adduserDataPart()` to trigger this behavior, optionally gated by a `makeDefault?: boolean` parameter (which defaults to `true` if no default has been set yet and defaults to `false` if a default has already been set).

This additional method also feels awkward.

File: packages/@aws-cdk/aws-ec2/lib/user-data.ts [hunk 2]
```diff
@@ -436,13 +436,15 @@ export interface MultipartUserDataOptions {
  *
  */
 export class MultipartUserData extends UserData {
-  private static readonly USE_PART_ERROR = 'MultipartUserData does not support this operation. Please add part using addPart.';
+  private static readonly USE_PART_ERROR = 'MultipartUserData does not support this operation. Please use addUserDataPartForCommands().';
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-ec2/lib/user-data.ts:439]: The error message could be more helpful about the behavior. How about:

```
MultipartUserData only supports this operation if it has a default UserData. Call addUserDataPart with makeDefault=true
```

**Review by rix0rrr — CHANGES_REQUESTED** [review]

**Comment by ddneilson** [comment]
Just fyi, @rix0rrr , the changes have been made & this is ready for you to take another pass at it.

**Comment by ddneilson** [comment]
@rix0rrr Just a reminder poke that this is waiting on you.

**Review by rix0rrr — APPROVED** [review]

**Pull request merged**